_delphi_utils_python/delphi_utils/flash_eval/eval_day.py

```diff
@@ -0,0 +1,415 @@
+    dist_min = pd.read_csv(f"flash_ref/{signal}/dist_min.csv")['0']
+    dist_max = pd.read_csv(f"flash_ref/{signal}/dist_max.csv")['0']
+    bef_list = list(np.zeros(7))
+    aft_list = list(np.zeros(7))
+    for i in range(7):
+        bef_list[i] = pd.read_csv(f"flash_ref/{signal}/iqr0_{lag}_{i}.csv")
+        aft_list[i] = pd.read_csv(f"flash_ref/{signal}/iqr1_{lag}_{i}.csv")
+    iqr_dict = {'Before': bef_list,
+                'After': aft_list}
+    weekday_params = pd.read_csv(f"flash_ref/{signal}/weekday_{lag}.csv", index_col=0)
+    linear_coeff = pd.read_csv(f"flash_ref/{signal}/linear_coeff_{lag}.csv", index_col=0)
+
+
+    # Make corrections & predictions
+    y_values_df, preprocess_outlier = eval_day(input_df, iqr_dict,
+                                               ref_date, weekday_params, linear_coeff)
+    s_val = y_values_df['y_raw'].to_frame()
+    out_range_outlier = pd.concat([s_val[s_val.y_raw < range_tup[0]],
+                                   s_val[s_val.y_raw > range_tup[-1]]], axis=0)
+
+    # Anomaly Detection
+    thresh = 0.01
+
+    val_min = return_vals(y_values_df, dist_min)[["pval"]]
+    val_max = return_vals(y_values_df, dist_max)[["pval"]]
+    val_min['flags'] = 'EVD_min'
+    val_max['flags'] = 'EVD_max'
+    val_min.columns = ['pval', 'flags']
+    val_max.columns = ['pval', 'flags']
+
+
+    min_thresh = thresh * 2
+    max_thresh = 1 - (thresh * 2)
+    max_anomalies = process_anomalies(val_max, 'max').dropna(
+        axis=1)
+    min_anomalies = process_anomalies(val_min, 'min').dropna(
+        axis=1)
+    max_anomalies = max_anomalies[max_anomalies.pval
+                                  > max_thresh].sort_values('pval')#.reset_index(drop=True)
+    min_anomalies = min_anomalies[min_anomalies.pval
+                                  < min_thresh].sort_values('pval')#.reset_index(drop=True)
+
+    starter_link = f"{HTML_LINK}{ref_date.strftime('%Y-%m_%d')},{report_date.strftime('%Y-%m_%d')}"
+
+    total_flags = 0
+    for (df, name) in zip(
+            [out_range_outlier, preprocess_outlier, max_anomalies, min_anomalies],
+            ['out_of_range', 'large_spike or weekday', 'max_anomalies', 'min_anomalies']):
+        p_text = ""
+        p_text += f"*{ref_date}* \n"
+        iter_df = df.copy()
+        if df.shape[0] > 20:
+            iter_df = iter_df.iloc[:20, :]
+        if iter_df.shape[0] > 0 :
+            for _, row in iter_df.reset_index().iterrows():
+                total_flags += 1
+                start_link = f"{starter_link},{row['index']}"
+                if 'pval' in iter_df.columns :
+                    p_text += f"\t{start_link}|*{row['index']}, {row.pval}*>\n"
+                elif 'y_raw' in iter_df.columns :
+                    p_text += f"\t{start_link}|*{row['index']}, {row.y_raw}*>\n"
+            logger.info(name,
+                        payload=p_text,
+                        hits=iter_df.shape[0])
+            p_text = ""
+
+
+def flash_eval(params):
```

Comment:
Recommend moving the contents of this function into `run_module` instead. Requires that `run_module` takes `params` as an argument.